packages/imperative/src/plugins/utilities/PluginIssues.ts

```diff
@@ -19,194 +19,194 @@ import { readFileSync } from "jsonfile";
+    /**
+     * This is the variable that stores the specific instance of the PluginIssues.
+     * Defined as static so that it can be accessed from anywhere.
+     *
+     * @private
+     * @type {PluginIssues}
+     */
+    private static mInstance: PluginIssues;
+
+    /**
+     * A map containing issues  for each plugin for which problems were detected.
+     *
+     * @private
+     * @type {IPluginIssues}
+     */
+    private pluginIssues: IPluginIssues = {};
+
+    /**
+     * The set of installed plugins. We access this class property only
+     * through its accompanying function
+     * [getInstalledPlugins]{@link PluginIssues#getInstalledPlugins}
+     * to ensure that we only read the file once and reduce excessive I/O.
+     *
+     * @private
+     * @type {IPluginJson}
+     */
+    private installedPlugins: IPluginJson = null;
+
+    // ___________________________________________________________________________
+    /**
+     * Gets a single instance of the PluginIssues. On the first call of
+     * PluginIssues.instance, a new Plugin Issues object is initialized and returned.
+     * Every subsequent call will use the one that was first created.
+     *
+     * @returns {PluginIssues} The newly initialized PMF object.
+     */
+    public static get instance(): PluginIssues {
+        if (this.mInstance == null) {
+            this.mInstance = new PluginIssues();
+        }
+
+        return this.mInstance;
     }
 
-    return this.mInstance;
-  }
-
-  // ___________________________________________________________________________
-  /**
-   * Reports whether or not a plugin has any issues with any of the specified
-   * severities.
-   *
-   * @param {string} pluginName - The name of the plugin
-   *
-   * @param {string} issueSevs - An array of issue severities.
-   *
-   * @returns {boolean} - True if any plugin issues have any of the severities.
-   *                      False otherwise.
-   */
-  public doesPluginHaveIssueSev(pluginName: string, issueSevs: IssueSeverity[]): boolean {
-    if (this.pluginIssues.hasOwnProperty(pluginName)) {
-      for (const nextSev of issueSevs) {
-        for (const nextIssue of this.pluginIssues[pluginName].issueList) {
-          if (nextIssue.issueSev === nextSev) {
-            return true;
-          }
+    // ___________________________________________________________________________
+    /**
+     * Reports whether or not a plugin has any issues with any of the specified
+     * severities.
+     *
+     * @param {string} pluginName - The name of the plugin
+     *
+     * @param {string} issueSevs - An array of issue severities.
+     *
+     * @returns {boolean} - True if any plugin issues have any of the severities.
+     *                      False otherwise.
+     */
+    public doesPluginHaveIssueSev(pluginName: string, issueSevs: IssueSeverity[]): boolean {
+        if (Object.prototype.hasOwnProperty.call(this.pluginIssues, pluginName)) {
```

Comment:
Just curious - what is the ESLINT rationale for using the Object.prototype form of hasOwnProperties?

Reply:
I think the reason they give is that calling `x.hasOwnProperty` directly on an object could be vulnerable to someone maliciously overriding the behavior of `hasOwnProperty` for the object `x`. I don't know how much we care about this, but it wasn't too hard to "fix" so I followed the ESLint recommendation 🙂 For more details about their rationale, see https://eslint.org/docs/rules/no-prototype-builtins.